python/lsst/summit/utils/peekExposure.py

```diff
@@ -0,0 +1,1026 @@
+            Mode run in.
+        binnedSourceCat : `lsst.afw.table.SourceCatalog`
+            Source catalog from the binned exposure.
+
+        Returns
+        -------
+        wgood : `numpy.ndarray`
+            Boolean array indicating which sources are good.
+        """
+        # Perform any filtering on the source catalog
+        wgood = np.ones(len(binnedSourceCat), dtype=bool)
+        if mode == "spec":
+            fpShapes = [src.getFootprint().getShape() for src in binnedSourceCat]
+            # Filter out likely spectrum detections
+            wgood &= np.array(
+                [sh.getIyy() < 10 * sh.getIxx() for sh in fpShapes], dtype=bool
+            )
+        return wgood
+
+    def getBrightest(self, binnedSourceCat, binSize, wgood):
+        """Find the brightest source in the catalog.
+
+        Parameters
+        ----------
+        binnedSourceCat : `lsst.afw.table.SourceCatalog`
+            Source catalog from the binned exposure.
+        binSize : int
+            Binning factor used.
+        wgood : `numpy.ndarray`
+            Boolean array indicating which sources are good.
+
+        Returns
+        -------
+        maxFluxIdx : int
+            Index of the brightest source in the catalog.
+        b_centroid : `lsst.geom.Point2D`
+            Centroid of the brightest source (unbinned coords).
+        b_shape : `lsst.geom.Quadrupole`
+            Shape of the brightest source (unbinned coords).
+        """
+        if np.sum(wgood) > 0:
+            fluxes = np.array([source.getApInstFlux() for source in binnedSourceCat])
+            idxs = np.arange(len(binnedSourceCat))
+
+            fluxes = fluxes[wgood]
+            idxs = idxs[wgood]
+
+            if np.sum(np.isfinite(fluxes)) > 0:
+                maxFluxIdx = idxs[np.nanargmax(fluxes)]
+                brightest = binnedSourceCat[maxFluxIdx]
+
+                # Convert binned coordinates back to original unbinned
+                # coordinates
+                b_x, b_y = brightest.getCentroid()
+                b_x = binSize * b_x + (binSize - 1) / 2
+                b_y = binSize * b_y + (binSize - 1) / 2
+                b_centroid = Point2D(b_x, b_y)
+                b_ixx = brightest.getIxx() * binSize**2
+                b_ixy = brightest.getIxy() * binSize**2
+                b_iyy = brightest.getIyy() * binSize**2
+                b_shape = Quadrupole(b_ixx, b_iyy, b_ixy)
+
+                return maxFluxIdx, b_centroid, b_shape
+        maxFluxIdx = IDX_SENTINEL
+        b_centroid = Point2D(np.nan, np.nan)
+        b_shape = Quadrupole(np.nan, np.nan, np.nan)
+        return maxFluxIdx, b_centroid, b_shape
+
+    def getPsfShape(self, binnedSourceCat, binSize, wgood):
+        """Estimate the modal PSF shape from the sources.
+
+        Parameters
+        ----------
+        binnedSourceCat : `lsst.afw.table.SourceCatalog`
+            Source catalog from the binned exposure.
+        binSize : int
```

Comment:
And again. It's systematic, so maybe you know something I don't 🙂

Reply:
I'm sure I don't. :)